shared/indicators.ts

```diff
@@ -0,0 +1,178 @@
+import { Base64 } from 'js-base64'
+import * as sourcegraph from 'sourcegraph'
+
+export const linkURL = 'https://docs.sourcegraph.com/code_intelligence/explanations/precise_code_intelligence'
+
+// Aggregable badges
+//
+// These indicators are the new hotness, and are tagged on each result that we send
+// back (except for LSP as we're still trying to sunset all of that related code).
+// These will be displayed aggregated at the file level to determine result precision.
+
+// TODO - remove once package is updated
+export interface AggregableBadge {
+    text: string
+    linkURL?: string
+    hoverMessage?: string
+}
+
+export const semanticBadge: AggregableBadge = {
+    text: 'semantic',
+    linkURL,
+    hoverMessage: "This data comes from a pre-computed semantic index of this project's source.",
+}
+
+export const searchBasedBadge: AggregableBadge = {
+    text: 'search-based',
+    linkURL,
+    hoverMessage: 'This data is generated by a heuristic text-based search.',
+}
+
+export const partialHoverNoDefinitionBadge: AggregableBadge = {
+    text: 'partial semantic',
+    linkURL,
+    hoverMessage:
+        'It looks like this symbol is defined in another repository that does not have a pre-computed semantic index. Go to definition may be imprecise.',
+}
+
+export const partialDefinitionNoHoverBadge: AggregableBadge = {
+    text: 'partial semantic',
+    linkURL,
+    hoverMessage:
+        'It looks like this symbol is defined in another repository that does not have a pre-computed semantic index. This hover text may be imprecise.',
```

Comment:
not sure if the right place for convo about this text is elsewhere, but 
`s/a pre-computed semantic index/semantic indexing/g`?

Reply:
This was your text! We can change it to whatever we like:

- It looks like this symbol is defined in another repository that does not have a pre-computed semantic index.
- It looks like this symbol is defined in another repository that does not have semantic indexing.
- It looks like this symbol is defined in another repository that does not have semantic indexing enabled.
- It looks like this symbol is defined in another repository that is not being semantically indexed.

Reply:
cc @AlicjaSuska